docs/source/developer_guide/testing_framework.rst

```diff
@@ -0,0 +1,269 @@
+* ``scenario`` objects, which encodes data inputs and method call sequences to ``estimator_instance`` (explained in further detail below).
+
+The ``scenario.run`` command is equivalent to calling ``estimator_instance.fit(**scenario_kwargs)``,
+where the ``scenario_kwargs`` are generated by the ``scenario``.
+
+It should be noted that the test is not decorated with fixture parametrization,
+the fixtures are instead generated by ``pytest_generate_tests``.
+
+The reason for this is that the applicable scenarios (fixture values of ``scenario``) depend on the ``estimator_instance`` fixture,
+since inputs to ``fit`` of a classifier will differ to an input to ``fit`` of a forecaster.
+
+Parameterized fixtures
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. _pytestfixtparam: https://docs.pytest.org/en/6.2.x/parametrize.html
+
+``sktime`` uses ``pytest`` fixture parameterization to execute tests in a loop over fixtures,
+for instance running all interface compatibility tests for all estimators.
+See the `pytest documentation on fixture parameterization <pytestfixtparam>`_ for an explanation of fixture parameterization.
+
+Currently, the ``sktime`` testing framework parameterizes the following fixtures in module level tests:
+
+* ``estimator``: all estimator classes, inheriting from the base class of the given module.
+In ``test_all_estimators``, this loops over all estimators.
+* ``estimator_instance``: all estimator test instances, obtained from all ``sktime`` estimators via ``create_test_instances_and_names``
+* ``scenario``: test scenarios, applicable to ``estimator`` or ``estimator_instance``.
+   The scenarios are specified in ``utils/_testing/scenarios_[estimator_scitype]``.
+
+Further parameterization may happen for individual tests, the scope is usually explained in the test docstrings.
+
+Scenarios
+~~~~~~~~~
+
+The ``scenario`` fixtures contain arguments for method calls, and a sequence for method calls.
+
+An example scenario specification, from ``utils/_testing/scenarios_forecasting``:
+
+.. code-block::
+
+   class ForecasterFitPredictUnivariateNoXLateFh(ForecasterTestScenario):
+      """Fit/predict only, univariate y, no X, no fh in predict."""
+
+      _tags = {"univariate_y": True, "fh_passed_in_fit": False}
+
+      args = {
+         "fit": {"y": _make_series(n_timepoints=20, random_state=RAND_SEED)},
+         "predict": {"fh": 1},
+      }
+      default_method_sequence = ["fit", "predict"]
+
+The scenario ``ForecasterFitPredictUnivariateNoXLateFh`` encodes instructions
+applied to an ``estimator_instance``, via instances ``scenario``.
+A call ``result = scenario.run(estimator_instance)`` will:
+
+1. first, call ``estimator_instance.fit(y=_make_series(n_timepoints=20, random_state=RAND_SEED))``
+2. then, call ``estimator_instance.predict(fh=1)`` and return the  output too ``result``.
+
+The abstraction of "scenario" allows to easily specify multiple argument combinations across multiple methods.
```

Comment:
Good practice is not to assume how easy/difficult things will be for the end user
```suggestion
The abstraction of "scenario" allows to specify multiple argument combinations across multiple methods.
```